---
title: "GSCA"
date: "`r BiocStyle::doc_date()`"
package: "`r BiocStyle::pkg_ver('GSCA')`"
output:
  BiocStyle::html_document2:
    fig_caption: true
    toc_float:
      collapsed: False
      toc_depth: 3
author:
- name: Andrew L. Thurman
- name: YounJeong Choi
- name: Christina Kendziorski
  email: kendzior@biostat.wisc.edu
vignette: >
  %\VignetteIndexEntry{GSCA}
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteEncoding{UTF-8}  
---
```{r knitsetup, message=FALSE, warning=FALSE, include=FALSE}
knitr::opts_knit$set(root.dir = ".")
knitr::opts_chunk$set(collapse = TRUE, warning = TRUE, fig.wide = TRUE)
BiocStyle::markdown()
library("BiocStyle")
```

# Introduction

`LungCancer3`in the package includes matched data matrices
from the three lung cancer microarray experiments, along with
the gene set definitions (in Entrez Gene ID) and their descriptions.

Gene matching by Entrez Gene ID provides 3,924 common genes across
the 3 studies of interest. Expression matrix rows are named in Entrez Gene IDs.
Multiple transcripts representing one ID have been averaged at the log level
before matching.

```{r}
library(GSCA, quietly = TRUE)
data(LungCancer3)
str(LungCancer3$data)
str(LungCancer3$info$GSdef[1:3])
str(LungCancer3$info$Name)
```


The `info`portion in `LungCancer3`contains gene set definitions
`GSdef`as a list of 3,649 gene sets. Each entry in the list is
a character vector of Entrez Gene IDs that belong to the corresponding
gene set. A particular set can be pulled out, for example,
GO:0008033 as shown here.

```{r}
GS <- LungCancer3$info$GSdef
GSdesc <- LungCancer3$info$Name
setid <- "GO:0008033"
GS[[setid]]
GSdesc[setid]
```


# Single-study differential correlation


The function `singleDC`calculates a dispersion index between
the tumor and normal groups for every gene set within a study.
It also calculates permutation-based $p$-values based on the given number
of sample permutations.

For example, if the Michigan study is of interest, one can do the following.
We try with 3 permutations to save time.

```{r}
dc.M <- singleDC(data = LungCancer3$data$Michigan,
                 group = c(86, 10), GSdefList = GS, nperm = 3)
str(dc.M)
```

# Meta-analysis of multiple studies

The function `metaDI`calculates a dispersion index between
two studies for every gene set. We choose to use the sign of
correlation difference between tumor and normal groups from each study here,
but the function itself can be used with the raw correlations as well.
It also calculates permutation-based $p$-values based on the given number
of permutations. The permutation here is annotated gene pair permutation.
For example, if the Harvard and Michigan studies are of interest,
one can do the following. We try with 3 permutations to save time.

```{r}
data.grouped <- list(Tumor =
                     list(Harvard = LungCancer3$data$Harvard[, 1:139],
                          Stanford = LungCancer3$data$Stanford[, 1:41],
                          Michigan = LungCancer3$data$Michigan[, 1:86]),
                     Normal =
                     list(Harvard = LungCancer3$data$Harvard[, 140:156],
                          Stanford = LungCancer3$data$Stanford[, 42:46],
                          Michigan = LungCancer3$data$Michigan[, 87:96]))
corr.t <- lapply(data.grouped$Tumor, function(x){
  cor(t(x), use = "pairwise.complete.obs")})

corr.n <- lapply(data.grouped$Normal, function(x){
  cor(t(x), use = "pairwise.complete.obs")})
cor.diff <- list(Harvard = corr.t$Harvard - corr.n$Harvard,
                 Michigan = corr.t$Michigan - corr.n$Michigan)
cor.diff.sign <- list(Harvard =
                      apply((cor.diff$Harvard > 0), 2, as.numeric) -
                      apply((cor.diff$Harvard < 0), 2, as.numeric),
                      Michigan =
                      apply((cor.diff$Michigan > 0), 2, as.numeric) -
                      apply((cor.diff$Michigan < 0), 2, as.numeric))
for (i in 1:length(cor.diff.sign)){
  rownames(cor.diff.sign[[i]]) <- colnames(cor.diff.sign[[i]])
}

dist.HM <- metaDI(cor.diff.sign$Harvard, cor.diff.sign$Michigan,
                  GS, 3, permDI = TRUE)
str(dist.HM)
```

# Network display

## Single-study network

`plotNW` is a wrapper function of `plot.graph` in `Rgraphviz`
package with specific settings. Nodes represent genes and edges represent
relationships between genes. Edges are colored in from blue to red
indicating correlation value of -1 to 1. This way we do not need to
threshold correlation values for drawing edges.
One can also choose to color nodes according to
the corresponding genes' DE information. Arguments that are accepted
by `plot.graph` can be passed. Figure \@ref{fig:nw} below
is an example with the set GO:0019216 drawn for the Michigan normal group,
with hypothetical DE (differential expression) information.

```{r nw, fig.width=7, fig.height=5, fig.cap="Network display of a gene set. Edge colors range from blue to red to indicate the correlation range of from -1 to 1. Differentially expressed gene nodes are colored in yellow."}
setid <- "GO:0019216"
gid <- GS[[setid]]
ss <- c("SERPINA3", "SOD1", "SCAP", "NPC2", "ADIPOQ",
        "PRKAA1", "AGT", "PPARA", "BMP6", "BRCA1")
# plotNW(genes = ss, cormatrix = corr.n$Michigan[gid, gid],
#        node.de = c(rep(1, 5), rep(0, 5)), ncolor = "yellow",
#        ecolor = bluered(201), ewidth = 5, gtype = "circo")
```

## Multi-edge network

The function `plotMNW`places nodes on a circle boundary and
connects every pair of nodes with multiple edges corresponding to
multiple studies of interest. Figure \@ref{fig:mnw} below is an example
of GO:0008033 with three studies.

```{r mnw, fig.width=7, fig.height=7, fig.cap="Multi-edge network display. Three lines per edge represent three study-specific correlations. Edges with similar color across studies indicate study agreement for the corresponding gene pair. Colors range from blue to red to indicate the correlation range of from -1 to 1."}
setid <- "GO:0008033"
gid <- GS[[setid]]
ss <- c("KARS", "SARS", "AARS", "SSB", "POP1", "RPP30")
cormatrix.list <- list(corr.n$Harvard[gid, gid], corr.n$Stanford[gid, gid],
                       corr.n$Michigan[gid, gid])
plotMNW(cormatrix.list, genes = ss, mycolors = bluered(201),
        ncolor = "yellow", node.de = c(rep(1, 3), rep(0, 3)),
        lwd = 5, jt = .3)
```
